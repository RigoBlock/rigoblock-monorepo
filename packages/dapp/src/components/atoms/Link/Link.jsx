import React from 'react'
import { Link as RouterLink } from 'react-router-dom'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import './Link.scss'

const Link = props => {
  const { size, children, to } = props
  const classProps = classNames('link', `${size}-link`)
  return (
    <RouterLink to={to} className={classProps}>
      {children}
    </RouterLink>
  )
}

Link.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.string.isRequired,
  size: PropTypes.string.isRequired
}

Link.defaultProps = {
  size: 'medium'
}

export default Link
