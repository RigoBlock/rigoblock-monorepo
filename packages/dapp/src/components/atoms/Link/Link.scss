@import "../../_settings/variables.scss";

.link {
  text-decoration: none;
  color: black;
}

.link-small {
  font-size: $base-font-size;
}

.link-medium {
  font-size: 1.3 * $base-font-size;
}

.link-large {
  font-size: 1.6 * $base-font-size;
}
