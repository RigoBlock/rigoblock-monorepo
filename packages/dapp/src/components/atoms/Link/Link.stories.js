import '../../_settings/_base.scss'
import { MemoryRouter } from 'react-router-dom'
import { select, text, withKnobs } from '@storybook/addon-knobs/react'
import { storiesOf } from '@storybook/react'
import Link, { LINK_SIZES } from './Link'
import React from 'react'

storiesOf('Atoms/Link', module)
  .addDecorator(story => <MemoryRouter>{story()}</MemoryRouter>)
  .addDecorator(withKnobs)
  .add('default', () => (
    <Link
      size={select('Link size', LINK_SIZES, LINK_SIZES.SMALL)}
      to={text('Links to', '/')}
    >
      Example Link
    </Link>
  ))
