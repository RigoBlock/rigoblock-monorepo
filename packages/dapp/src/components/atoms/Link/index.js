import Link, { LINK_SIZES } from './Link'

export { LINK_SIZES }

export default Link
