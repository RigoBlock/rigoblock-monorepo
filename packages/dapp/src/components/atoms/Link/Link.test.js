import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Link from './Link.jsx'

const props = { to: '#', children: 'Test link' }

describe('Link component', () => {
  it('renders correctly', () => {
    expect(
      toJson(shallow(createComponentWithProps(Link, props)))
    ).toMatchSnapshot()
  })
})
