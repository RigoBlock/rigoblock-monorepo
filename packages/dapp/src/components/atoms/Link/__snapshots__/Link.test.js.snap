// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component renders correctly 1`] = `
<Link
  className="link medium"
  replace={false}
  to="#"
>
  Test link
</Link>
`;
